src/Common/src/CoreLib/System/IO/Stream.cs

```diff
@@ -758,17 +758,13 @@ public virtual int Read(Span<byte> buffer)
 
         // Reads one byte from the stream by calling Read(byte[], int, int). 
         // Will return an unsigned byte cast to an int or -1 on end of stream.
-        // This implementation does not perform well because it allocates a new
-        // byte[] each time you call it, and should be overridden by any 
-        // subclass that maintains an internal buffer.  Then, it can help perf
-        // significantly for people who are reading one byte at a time.
         public virtual int ReadByte()
         {
-            byte[] oneByteArray = new byte[1];
```

Comment:
Probably similar but:

```C#
byte b = 0;
Span<byte> oneByteSpan = MemoryMarshal.CreateSpan(ref b, 1);
int r = Read(oneByteSpan);
if (r == 0)
    return -1;
return b;
```

Reply:
https://github.com/dotnet/corefx/pull/36838#discussion_r275114997
Thoughts?

Reply:
Yes it makes things worse in the cases where `Read(Span<byte>)` isn't overridden. That said, I don't think overriding ReadByte is common (I know none of our derived streams do it today). I also don't know what actual code calls ReadByte 😄

Reply:
Ah alright, was not aware, will revert the changes!

Reply:
Reverted in https://github.com/dotnet/corefx/pull/36838/commits/1daf838d3f5a474856e9c3503604ba37758ad8fc